### ISSUE TITLE
CRLF linebreaks treated as two individual line breaks

### ISSUE THREAD

**Issue by Gipphe** [plaid/transcribe#23]
A file with `CRLF` (windows-style) newlines, instead of just `LF` (linux-style), is treated as having two newlines per `CRLF` pair in code blocks, instead of just one newline.

Example source file:
```javascript
//# identity :: a -> a
//.
//. The identity function
//.
//. ```javascript
//. > identity(1)
//. 1
//. ```
function identity(x) {
	return x;
}
```
---------
Actual output, with `CRLF` newlines:

````markdown
### <a name="identity" href="'https://github.com/Gipphe/transcribe-cr-lf-issue/blob/master/index.js#L1'">`identity :: a -⁠> a`</a>

The identity function

```javascript

> identity(1)

1

```
````
---------
Expected output, which you only get with `LF` newlines:

````markdown
### <a name="identity" href="'https://github.com/Gipphe/transcribe-cr-lf-issue/blob/master/index.js#L1'">`identity :: a -⁠> a`</a>

The identity function

```javascript
> identity(1)
1
```
````
--------

A minimal repo with the issue in question is available at [here](https://github.com/Gipphe/transcribe-cr-lf-issue).

**Comment by davidchambers** [comment]
Thanks for the terrific bug report, @Gipphe!

I've identified the [problematic code][1]:

```javascript
> R.match (/^.*$/gm) ('foo\nbar\nbaz\n')
['foo', 'bar', 'baz', '']

> R.match (/^.*$/gm) ('foo\r\nbar\r\nbaz\r\n')
['foo', '', 'bar', '', 'baz', '', '']
```

We need to replace all occurrences of `'\r\n'` with `'\n'` before matching. I'll submit a pull request.


[1]: https://github.com/plaid/transcribe/blob/v1.1.1/bin/transcribe#L75

**Comment by Gipphe** [comment]
Absolutely wonderful!

**Comment by davidchambers** [comment]
I just merged #25 and released [v1.1.2][1]. I'm sorry for taking so many months to do so.

Thanks again for your contribution, @Gipphe. :D


[1]: https://github.com/plaid/transcribe/releases/tag/v1.1.2
